Extended docs with logging customisation at runtime (#3968)
kedro-org/kedro: docs/source/logging/index.md

If the `KEDRO_LOGGING_CONFIG` environment variable is not set, Kedro will use the [default logging configuration](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/default_logging.yml).
```

> One can also customise logging at the runtime and redefine the logging configuration provided in the `logging.yml`.

Reviewer: ```suggestion

### Change the verbosity of specific parts of Kedro

One can also customise logging at the runtime and redefine the logging configuration provided in the `logging.yml`.
```

Reviewer 2: "You can" instead of "One can", maybe? Most of the docs are in second person I think

> In the example below we change a logging level to `WARNING` for the `kedro.io.data_catalog` component, so the setting is only applied to the `kedro.io.data_catalog` logger and for the rest of the components the default settings provided via `logging.yml` are remained unchanged.

Reviewer: ```suggestion
In the example below we change a logging level to `WARNING` for the `kedro.io.data_catalog` component logger specifically, the logging for the rest of the components will remain unchanged.
```

If the `KEDRO_LOGGING_CONFIG` environment variable is not set, Kedro will use the [default logging configuration](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/default_logging.yml).
```

### Change the verbosity of specific parts of Kedro

One can also customise logging at the runtime and redefine the logging configuration provided in the `logging.yml`.
In the example below we change a logging level to `WARNING` for the `kedro.io.data_catalog` component logger specifically, the logging for the rest of the components will remain unchanged.
The same can be done for higher/lower-level components without affecting the top-level.

> ```python

Author: That's a good catch, I've added a clarification that this is relevant for the notebook environment. We cannot make it in `settings.py` since logging configuration from `logging.yml` is applied after `settings.py` configuration https://github.com/kedro-org/kedro/blob/413dbca46a9cb8e29fc1761564c4e13e2efe4d8e/kedro/framework/project/__init__.py#L273, so it will be redefined by what is provided in `logging.yml`

> ```python

Author: So, the alternative is to customise `logging.yml` described in the neighbouring section.

> ```python

Reviewer 2: Ah I see, makes sense!

If the `KEDRO_LOGGING_CONFIG` environment variable is not set, Kedro will use the [default logging configuration](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/default_logging.yml).
```

### Change the verbosity of specific parts of Kedro

> You can also customise logging at the runtime and redefine the logging configuration provided in the `logging.yml` when using jupyter notebook.

Reviewer 3: ```suggestion
You can also customise logging at runtime and redefine the logging configuration provided in the `logging.yml` when using jupyter notebook.
```

> In the example below we change a logging level to `WARNING` for the `kedro.io.data_catalog` component logger specifically, the logging for the rest of the components will remain unchanged.

Reviewer 3: ```suggestion
The example below demonstrates how you can change the logging level from `INFO` to `WARNING` for the `kedro.io.data_catalog` component logger specifically, the logging for the rest of the components will remain unchanged.
```
